### DISCUSSION TITLE
Preventing player movement before chunks loaded

### DISCUSSION THREAD

**Discussion by FluxCapacitor2** [Minestom/Minestom#1403] [Help]
When a player joins their spawning instance (set in PlayerLoginEvent), if the chunk takes enough time to load, they can fall a significant distance. Is there a way to prevent the client's movement until they have received a chunk, or at least to know when a client receives a chunk? I suppose you could listen for the chunk packets using the PlayerPacketOutEvent, but I am wondering if there is a better solution than that.

A solution to this would be useful for instances with slow chunk generators/loaders and for minigames which kill the player when they touch the void.

**Comment by KodingDev** [comment]
Relevant discussion on Discord that was had before for some context (dev wise)

<img width="824" alt="CleanShot 2022-09-13 at 11 24 08@2x" src="https://user-images.githubusercontent.com/23714764/189786869-bbf4cb24-06eb-46ad-a5d1-98a0ea97eab4.png">
<img width="832" alt="CleanShot 2022-09-13 at 11 24 38@2x" src="https://user-images.githubusercontent.com/23714764/189786900-af888a96-8a56-421f-8bd1-b3d29fcfe2e5.png">
